animal.go

```diff
@@ -17,6 +18,51 @@ const (
 	AnswerLose = "I lose!!"
 )
 
+// Question ...
```

Comment:
We're in a good place to add some documentation comments now, aren't we? I think the design has more or less stabilised enough.

Reply:
Added comments in the updated commit.